### ISSUE TITLE
Add support for policy changesets

### ISSUE THREAD

**Issue by schrockwell** [schrockwell/bodyguard#1]
It might be cool if policies could also provide a "starting point" changeset. Maybe include `cast` operations, but no validations necessarily. Basically it would just be used to whitelist parameters passed in to a changeset.

The rough idea:

``` elixir
defmodule Post.Policy do
  def changeset(user, _action, post, params, _opts) do
    cond do
      # Admins can change title and body
      user.role == :admin -> 
        cast(post, params, [:title, :body])

      # Owners can only change title
      user.role == :user && user.id == post.user_id -> 
        cast(post, params, [:title])

      # Nobody else can change anything
      true ->
        cast(post, params, [])
    end
  end
end
```

**Comment by bencates** [comment]
This is pretty easy (via [`Ecto.Changeset.apply_changes/1'](https://hexdocs.pm/ecto/Ecto.Changeset.html#apply_changes/1)), but that does require pulling in Ecto as a dependency.

[Apparently Mix supports optional dependencies.](https://groups.google.com/forum/#!topic/elixir-lang-talk/Xe9uXE3uGsY)  I'm gonna try playing with that to see if it works for our use case.

**Comment by JoshSmith** [comment]
Hey @schrockwell, any movement on this at all? We've implemented a `load_and_authorize_changeset` plug that works with Canary.

**Comment by schrockwell** [comment]
Hey @JoshSmith! No we haven't made any progress on this feature yet. I'd be curious to see what you've got if you could share it.

**Comment by JoshSmith** [comment]
Sure, let us flesh some things out a bit and circle back here.

**Comment by schrockwell** [comment]
The more I think about this, the more I like borrowing `permitted_attributes` from Pundit. Help the Rails guys feel at home.

Instead of dealing with changesets directly (and adding a dependency on Ecto), just return arrays of atoms for the permitted attributes based on user. Then it's persistence-agnostic too. It's not nearly as flexible as  changesets, but it covers most use cases.

**Comment by schrockwell** [comment]
`permitted_attributes` available as of v0.3.0

**Comment by JoshSmith** [comment]
@begedin ☝️
